### ISSUE TITLE
Check latest CUBs for best compatibility for WiX v4 GA

### ISSUE THREAD

**Issue by barnson** [wixtoolset/issues#6567]
This appears to be a bug in the CUBs in v3.14.

### Discussed in https://github.com/wixtoolset/issues/discussions/6524

<div type='discussions-op-text'>

<sup>Originally posted by **mjanulaitis** July 21, 2021</sup>
* Which version of WiX are you building with?
> v3.14.0.4118

* Which version of Visual Studio are you building with (if any)?
> 16.9.3

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?
> NA

* Which version of .NET are you building with?
> 4.7.2

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?
> This is a build issue

* Describe the problem and the steps to reproduce it.
* After upgrading to the latest version I get the following errors when building.  Please note the merge module in question has not changed in over 6 months.

2>light.exe(0,0): error LGHT0204: ICEM10: The property 'ProductCode' is not allowed in a Merge Module
2>C:\Users\mike\AppData\Local\Temp\pg1e2w52\pg1e2w52\ConsoleMergeModule.msm(0,0): error LGHT0204: ICE03: Table: Component Column: _IceM05Mark Missing specifications in _Validation Table (or Old Database)
2>C:\Users\mike\AppData\Local\Temp\pg1e2w52\pg1e2w52\ConsoleMergeModule.msm(0,0): error LGHT0204: ICE03: Table: File Column: _ICEM07CAB Missing specifications in _Validation Table (or Old Database)
2>light.exe(0,0): error LGHT0217: Error executing ICE action 'ICE103'. The most common cause of this kind of ICE failure is an incorrectly registered scripting engine. See http://wixtoolset.org/documentation/error217/ for details and how to solve this problem. The following string format was not expected by the external UI message logger: "There is a problem with this Windows Installer package. A DLL required for this install to complete could not be run. Contact your support personnel or package vendor.  ".

* Describe the behavior you expected and how it differed from the actual behavior.
> I expected the build to complete successfully.
</div>

**Comment by barnson** [comment]
Running merge-module validation using Orca v5.0.22000.194 from the Windows 11 SDK v10.0.22000 on a well-formed merge module fails when executing ICE103:

    Action 18:39:33: ICE103. 
    Action start 18:39:33: ICE103.
    MSI (c) (34:E8) [18:39:33:886]: Invoking remote custom action. DLL: X:\Tmp\MSI3AEA.tmp, Entrypoint: ICE103
    CustomAction ICE103 returned actual error code 1154 (note this may not be 100% accurate if translation happened inside sandbox)
    MSI (c) (34:98) [18:39:33:890]: Note: 1: 1723 2: ICE103 3: ICE103 4: X:\Tmp\MSI3AEA.tmp 
    MSI (c) (34:98) [18:39:33:890]: Note: 1: 2205 2: X:\Tmp\ICE30AE.tmp 3: Error 
    MSI (c) (34:98) [18:39:33:890]: Note: 1: 2228 2: X:\Tmp\ICE30AE.tmp 3: Error 4: SELECT `Message` FROM `Error` WHERE `Error` = 1723 
    Error 1723. There is a problem with this Windows Installer package. A DLL required for this install to complete could not be run. Contact your support personnel or package vendor.  Action ICE103, entry: ICE103, library: X:\Tmp\MSI3AEA.tmp 

Attaching repro data for a bug report to Microsoft, should such a thing be possible.

[Repro6567.zip](https://github.com/wixtoolset/issues/files/7878227/Repro6567.zip)

**Comment by barnson** [comment]
https://github.com/microsoft/Windows-Dev-Performance/issues/104

**Comment by barnson** [comment]
We should consider downgrading cubs to the v3.11 versions, the last-known good set.

**Comment by chrpai** [comment]
I just came across this.  WiX 3.11.2 is fine but all of the 3.14 dev builds give ICE103.

**Comment by chrpai** [comment]
What's the plan for this issue?  I went back through history and see the changes were to update to support ARM64.  Is there a newer CUB from MSFT or something that will solve that problem and not have ICE103 validation errors?

**Comment by chrpai** [comment]
I just downloaded the latest windows sdk and installed orca and validation msis and compared to wix v3.  CUBs aren't versioned or signed as I can tell but the mergemod.dll currently commited is 5.0.18362.1 signed 3/18/2019.  The latest DLL on my machine is  5.0.22000.194 signed 9/12/2021.

I'm going to submit a PR containing the latest mergemod.dll,  darice.cub and mergemod.cub.  I'd suggest we do a development release and then regression test this issue plus the ARM64 issue that was originally being fixed.

**Comment by imag0r** [comment]
The issue is - ICE103 is defined as custom action from Shared15Ice binary. Exporting that binary and examining DLL exports reveals that ICE103 export is missing. The workaround I use for now is explicitly suppress ICE103 in my merge module projects.

**Comment by chrpai** [comment]
I'm not to keep on disabling validation and I'm also not too keep on modifying the .cub to remove ICE103 (although it could be done).  Anyone have any pull at MSFT these days to get them to fix it?

**Comment by imag0r** [comment]
> I'm not to keen on disabling validation and I'm also not too keen on modifying the .cub to remove ICE103 (although it could be done). 

You can suppress just that one validator. No need to disable all of them.

**Comment by chrpai** [comment]
I'm well aware of that.  Let me rephrase, I'm not keen on disabling any ICEs.  I've consulted for hundreds of clients over the years and have thousands of .wixprojs in repos that I can still see and can't see.  I wouldn't want someone upgrading to 3.14 to have a build break because of a dorked up .cub from MSFT.

**Comment by chrpai** [comment]
Visual Studio closed the issue and said file it through feedback hub.  I submitted it here.  Please upvote. It probably doesn't help but it can't hurt.

https://aka.ms/AAfvup7

Personally I'm thinking we just fork the cub and castrate the ICE.

**Comment by imag0r** [comment]
Upvoted.  Forking the .cub makes perfect sense too, at least temporarily.

**Comment by XomegaNet** [comment]
@chrpai fyi - I'd love to upvote, but when trying to open the above ticket in the feedback hub, I'm getting an error "Your account doesn't have access to this feedback".

**Comment by barnson** [comment]
Assuming that Microsoft doesn't fix the issue, what do we do in v3.x? in v4.0?

**Comment by barnson** [comment]
Use latest cubs that work: v3.14 version of darice.cub and 22000 build of mergemod.cub.

**Comment by barnson** [comment]
ICE103 is still broken in latest mergemod.cub. Plan: Update to latest *.cub and suppress ICE103 when validating *.msm. Question for triage: Do we implement ICE103 inside WiX and suppress it like the other ICEs we suppress because >= checks are built in? Or just let it go because it's for new-to-MSI5 UI stuff inside merge modules?

**Comment by chrpai** [comment]
This is still broken in v4.0.0.  Creating a merge module with a property that is marked Secure causes:

Severity	Code	Description	Project	File	Line	Suppression State
Error	WIX0204	ICEM10: The property 'ProductCode' is not allowed in a Merge Module	MergeModule1	C:\GitHub\EUCVA-Server\Installer\MergeModule1\wix.exe	1	
Error	WIX0204	ICE03: Table: Component Column: _IceM05Mark Missing specifications in _Validation Table (or Old Database)	MergeModule1	C:\GitHub\EUCVA-Server\Installer\MergeModule1\obj\x64\Debug\MergeModule1.msm	1	
Error	WIX0204	ICE03: Table: File Column: _ICEM07CAB Missing specifications in _Validation Table (or Old Database)	MergeModule1	C:\GitHub\EUCVA-Server\Installer\MergeModule1\obj\x64\Debug\MergeModule1.msm	1	


However if you look at the MSM in ORCA ProductCode is no where to be found and MSM validation works.
